### PR TITLE
Added support for Laravel 5.8

### PR DESCRIPTION
**Pull request by jabarrioss**
I fixed a bug when firing events, from laravel 5.5 to fire an evento you should use the Event::dipatch() method instead of the Event::Fire() method

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-    "name": "gloudemans/shoppingcart",
+    "name": "jabarrioss/shoppingcart",
```
Comment (bumbummen99) [inline composer.json:2]: Please remove this change (from the PR).

File: src/Cart.php [hunk 2]
```diff
@@ -47,7 +47,7 @@ public function __construct(SessionManager $session, Dispatcher $events)
     {
         $this->session = $session;
         $this->events = $events;
-
+
```
Comment (bumbummen99) [inline src/Cart.php:50]: Nit: Is this change required?

**Review by bumbummen99 — CHANGES_REQUESTED** [review]
Hey, please do not change the package name when you plan to create a PR. The proper way to handle this case would be to create an additional branch just for backporting/pr and keep the original package name there. Also do the tests still run properly?

**Comment by jabarrioss** [comment]
> Hey, please do not change the package name when you plan to create a PR. The proper way to handle this case would be to create an additional branch just for backporting/pr and keep the original package name there. Also do the tests still run properly?

I haven't run the test I changed the name of the package because I have to use for a project that I'm working on, and I needed to install it directly from my github

**Comment by bumbummen99** [comment]
@jabarrioss That is absolutely fine, no worries! :) But this PR simply can not be merged with a different package name and you can not simply change it as your project depends on it. That is why i suggested creating the pr with a different branch that still has the original package-name but all the other changes.

**Comment by rbergDrox** [comment]
@jabarrioss  @bumbummen99  here my PR #608  
php version for laravel  >=7.1.3
Logout event  signature has changed I fixed by adding web guard
I passed the tests

**Pull request closed without merging**